### PR TITLE
feat: Add Modal, SelectMenu, and Radio Components

### PR DESCRIPTION
**Pull request by jipatino98**
## Summary
This PR incorporates reusable UI components that can be used any where in our application. These components include:
1. Modal - a UI component that displays information, usually in front of and deactivates all other page content
2. SelectMenu - A component that displays a list of options that the user can choose from, which is triggered by a button
3. Radio - A set of checkable buttons (aka radio buttons) where no more than one of the buttons can be checked at a time.

These components were created using Radix Primitives, a headless component library that ensures functionality and accessibility for all web users. This includes keyboard navigation and support for screen readers.

A more detailed documentation on how to use these components and the various props they can take can found [here](https://cascarita.atlassian.net/wiki/spaces/~71202069510b02626642319e93c473c7e23d26/pages/32309250/Component+Documentation#Modal).

## Demo
<div>
    <a href="https://www.loom.com/share/478e5db3bd064accaaeb450921046ee9">
      
    </a>
  </div>

## Summary by CodeRabbit

- **New Features**
    - Introduced a `LeagueForm` component for creating new leagues, allowing users to input league details and link existing divisions.
- **Style**
    - Defined styles for the `LeagueForm` component, including form layout, input fields, radio buttons, and buttons for submission.

### CODE REVIEW

File: client/components/Forms/LeagueForm.module.css [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+.Label {
+  font-size: 1.1rem;
+  color: black;
+  margin-right: 16px;
+  display: block;
+}
+
+.Input {
```
Comment (abanuelo) [inline client/components/Forms/LeagueForm.module.css:21]: we should make css camelCase

**Review by abanuelo — APPROVED** [review]
Just a comment on naming conventions but overall LGTM great work!

File: client/components/Forms/LeagueForm.module.css (hunk 1 above)
Comment (Jen-464) [reply client/components/Forms/LeagueForm.module.css:21]: ^^

**Review by Jen-464 — APPROVED** [review]
Everything looks good!

File: client/components/Forms/LeagueForm.module.css (hunk 1 above)
Comment (jipatino98) [reply client/components/Forms/LeagueForm.module.css:21]: Got it! Addressed that feedback

**Pull request merged**